### ISSUE TITLE
Tag splitting not reflected in Paned Browser

### ISSUE THREAD

**Issue by 1xPdd** [quodlibet/quodlibet#2407]
Hi, a long-time user of Quod Libet, I really wanted to thank you for this great software.

I tried to read up on this and I'm fairly certain I've setup Quod Libet correctly. In the settings, I have '; / & ,' (without quotes) set as split indicators. However, in both the Genre and People Panes, multi-genre and multi-artists entires are not split into multiple entries. Thus, for instance, I have genre listings "French; Jazz," "French; Pop," and "French; Pop; Jazz." If I understand the tag-split feature correctly, there should be three genres, respectively, "Jazz," "French,"and "Pop." The same lack of splitting occurs with the people pane. Also, this problem doesn't seem to depend on which separator is used--as far as I can tell, none of the split indicators actually split the tags.

Thanks in advance,

**Comment by declension** [comment]
Hi @1xPdd. Thanks.

> Also, this problem doesn't seem to depend on which separator is used--as far as I can tell, none of the split indicators actually split the tags. 

Yes that's correct in fact. The _Split On_ preferences control the Edit Tags splitting feature (right-click on a tag value when editing to see). So you'll need to split the tags _permanently_ and save to your songs (internally QL will generally store these as newline-delimited string values). I recommend doing this, as things like `~people` will make more sense, and the Paned Browser will work much better. In fact I'd say QL's use of multi-valued tags is one of its most fundamental features, with knock-on effects in many aspects of the browsers and tagging.

But this confusion has definitely come up before (I raised something pretty similar in #1029 ages ago).  - perhaps we should update some documentation / tooltips / other UI. 

See also #1740 etc

**Comment by urielz** [comment]
Perhaps an option somewhere (or plugin) can be added to do a check on the entire database and split fields according to the Split On preferences? That way the user doesn't have to do it manually every time an entry is added to the database thus saving time...

**Comment by declension** [comment]
@urielz Yeah a plugin to do this might help. It's non-interactive anyway so could automate well, though I feel it often doesn't do "quite enough" or is more subtle than people want that I wouldn't dare run it myself, unless it collected all results to review (easily) before saving...

**Comment by zarfld** [comment]
i just ran into the same issue, and i wonder if i would have a possibility to do an outomation script for that task on my own. 
for me i would just want to filter Genre for "; " (which is my seperator), and then Split them up one by one. (whatever tasks need to be done what i need 3 Clicks using Browser/EditTag UI)

**Comment by budhajeewa** [comment]
Did anyone find a way to split tags in the view of the application, while still using inline separators (Delimiters) like ";"?

**Comment by declension** [comment]
This could be done in the Python console (plugin) fairly easily I imagine @budhajeewa  @zarfld
